### DISCUSSION TITLE
How to transform a fork to a repository?

### DISCUSSION THREAD

**Discussion by Marco-Sulla** [community/community#52663] [Repositories] [answered]
### Select Topic Area

Question

### Body

I started to work to https://github.com/Marco-Sulla/python-frozendict forking the original repo. Now I am the official maintainer on Pypi, and I started to realize that a fork has not the same visibility of a normal repo. It's not included in searches on Github and in Google too.

How can I transform the fork in a normal repo?

**Comment by airtower-luna** [comment, marked as answer]
There's a "detach fork" option in the support menu: https://support.github.com/request/fork

Clicking it opens a chatbot. I haven't had a need to use it so far, so I can only say that I hope it works. :wink:

**Reply by Marco-Sulla** [reply]
Thank you, ticket opened and finger crossed X-D

**Reply by jadurani** [reply]
Hey @Marco-Sulla 👋🏽  was the detachment successful on your end? how long did it take?

**Reply by Marco-Sulla** [reply]
Yes, and it was very quick.

**Reply by avylove** [reply]
@Marco-Sulla  Did it preserve all your history / releases / PRs / etc?

**Reply by Marco-Sulla** [reply]
@avylove Yep

**Reply by ilyhalight** [reply]
> @Marco-Sulla Did it preserve all your history / releases / PRs / etc?

When I asked them about this recently, they told me that only the commit history would be saved, and all issues, PR, stars, etc. would be lost 😞

**Reply by MaxymVlasov** [reply]
Just converted https://github.com/MaxymVlasov/dive-action from fork to standalone repo by `extracting` it as TC suggested in support menu 
All releases, stars, PRs and so on still in place :tada: 


And it takes less than a day to resolve
![image](https://github.com/user-attachments/assets/159f7c5c-690a-4b22-a1e7-1f52d6c9ccc5)

**Reply by ilyhalight** [reply]
> All releases, stars, PRs and so on still in place 🎉

Now, they really detach repo with save all stats :)
![firefox_bwjrKlfVer](https://github.com/user-attachments/assets/3b29b2c9-7d64-4d32-bf2f-4d48ee255ab6)

**Reply by merusso** [reply]
This should be a self-service option without involving GitHub Support.

I'm a GitHub Enterprise user and I'm the admin for the fork repo, but I'm not able to make this request directly because GitHub Support requires an organization admin to submit this request. So I have to ask admins to make this request on my behalf.

When using Bitbucket, this is a self-service change for repo admins.

**Comment by opegamanalitarok** [comment]
Unfortunately, you cannot directly convert a forked repository into a standalone repository on GitHub. However, you can turn your fork into a standalone repository by creating a new repository and copying the contents of your fork into it. Here's how you can do it:

1. Create a new repository on GitHub by clicking on the "New" button on your GitHub dashboard and giving it a name.
2. Clone your forked repository to your local machine using the git clone command.
3. Change your current directory to the cloned repository and remove the link to the original repository using the git remote remove command.
4. Add a new remote link to the repository you just created on GitHub using the git remote add command.
5. Push the changes to the new repository using the git push command.

Here's an example of the commands you can use:

# Clone your forked repository to your local machine
git clone https://github.com/yourusername/python-frozendict.git

# Change your current directory to the cloned repository
cd python-frozendict

# Remove the link to the original repository
git remote remove origin

# Add a new remote link to the repository you just created on GitHub
git remote add origin https://github.com/yourusername/new-repository.git

# Push the changes to the new repository
git push -u origin master

After you've pushed the changes to the new repository, you can delete the forked repository on GitHub. Note that this process will not preserve the forks, stars, or watchers of your original repository.

Hope this helps! 🙌🙌

**Reply by airtower-luna** [reply]
> Unfortunately, you cannot directly convert a forked repository into a standalone repository on GitHub.

This is plainly wrong, as I mentioned above. Sure, creating a new repository will give you a standalone repository, but you'll lose all issues, PRs, etc. *and* effectively kick out all watchers, people who starred the repo, and so on. So it's the worst possible option.

**Reply by mikekellyonline** [reply]
Random Followup for this...what if you have been working on a fork and the maintainer is absent for more than 2 years on merging any PR's? Would that be a reasonable cause to "takeover" the project?

**Reply by Lawlor-C** [reply]
Similar question - I forked a repo so I could complete a team project on my own. That repo was itself a fork from the education body we were studying in, so inherently Private. I want to make my fork Public and I'm the only one involved so watchers/issues etc is irrelevant.
In this instance, would creating the standalone repo actually be the best way to proceed?

**Comment by ryan-winkler** [comment]
Hey @Marco-Sulla @airtower-luna @airtower-luna @merusso @MaxymVlasov @jadurani 

I opened a product-feedback request proposing that GitHub preserve the relationship while separately recording original downstream contribution activity, and how forcing forks to detach from their upstream can cause harm and reduced visibility into issues.



https://github.com/orgs/community/discussions/205090

This discussion demonstrates an unintended incentive in GitHub's treatment of forks.

Detaching a repository can give it ordinary standalone treatment, but it also removes useful lineage. The code, authorship, maintenance work, and user value do not become more legitimate merely because the fork relationship is removed.



The core principle is:

> Do not attribute downstream work to upstream until upstream accepts it. Do still attribute the original work to the downstream contributor and repository where it occurred.

If you can comment how substantial work is being hidden due to this rule, or any use cases or needs to have this changes It will help strengthen the case.

My singular voice won't be enough to change it, but if we have multiple voices, that would be great.

**Reply by Marco-Sulla** [reply]
It was not my problem. My problem was there's no way to take the ownership of a repo like PyPI allows for their projects, and/or you can't signal a repo as abandoned and ask to mark it as archived.

Anyway, I like the idea. +1 :)
